test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -675,6 +675,20 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 					}, util.ConsistentDuration, util.Interval).Should(gomega.BeFalse())
 				})
 
+				ginkgo.By("checking that WaitingForReplacementPods status is set to true", func() {
```

Comment:
I would also like a stronger test case:

1. Evict the group
2. Observe the Pod to have a deletionTimestamp
3. Immediately readmit the workload
4. Observe that the workload has the condition for WaitingForReplacementPod set to true with reason related to the eviction.
5. Only now: create replacement pod
6. Observe condition WaitingForReplacementPod transition back to false

I think the existing implementation should pass this test, but the above is what worries the customers (that the Workload gets readmitted too fast and they miss the eviction condition).